docs/tls-authentication.md

```diff
@@ -0,0 +1,248 @@
+system.
+
+### Default (localhost, 3 years)
+
+#### Downloading Script
+
+ _[daedalus](https://github.com/input-output-hk/daedalus)_ provides a script
+ `installers/build-certificates-unix.sh` (or a windows equivalent
+ `installers/build-certificates-win64.bat`) to generate default certificates for both
+ _cardano-sl_ and _daedalus_, as well as a CA public certificate used to sign the formers. 
+
+Make sure you have an Internet access and download the installer script and the necessary
+configuration files as such:
+
+```
+$ wget -q https://raw.githubusercontent.com/input-output-hk/daedalus/master/installers/build-certificates-unix.sh
+$ wget -q https://raw.githubusercontent.com/input-output-hk/daedalus/master/installers/{ca,server,client}.conf
+```
+
+#### Running Script
+
+Run the script to generate the certificates as follows:
+
+```
+$ bash ./build-certificates-unix.sh 
+```
+
+Once done (`Oll Korrect`), a folder `tls` should have been created in your current directory
+containing, at least, the following files:
+
+```
+--| tls
+-----| ca
+--------| ca.crt
+-----| client
+--------| client.crt
+--------| client.key
+-----| server
+--------| server.crt
+--------| server.key
+```
+
+Those are the certificates and private keys that _daedalus_ and _cardano-sl_ will require to
+talk to each other. See [Using Certificates](#using-certificates) for more details.
+
+
+### Personalized 
+
+Follow the steps described in [Default | Downloading Script](#downloading-script) to first pull
+the build script and the required configuration files.
+
+> **NOTE:** You can also simply use compatible certificates that have been signed by an
+> existing authority. Make sure you have the public certificate of that authority in your 
+> possession.
+
+#### Custom Server Domains
+
+The `build-certificates-*` script will generate server's certificates for `localhost`. This
+seemingly assume several things:
+
+  - The backend nodes and frontend are running on the same machine
+  - You don't need to enforce the backend domain (multiple services may be running on `localhost`)
+
+In any case, you might need to run the backend on a different domain and therefore, identify it
+using something else than `localhost`. To do so, modify the `[ server_alt_names ]` section in
+the `ca.conf` downloaded earlier:
+
+```
+[ server_alt_names ]
+DNS.1                   = localhost.localdomain
+DNS.2                   = localhost
+DNS.3                   = 127.0.0.1
+DNS.4                   = ::1
+```
+
+and replace DNS references with your actual domain. For instance:
+
+```
+[ server_alt_names ]
+DNS.1                   = www.example.com
+DNS.2                   = example.com
+```
+
+Then, follow the steps described in [Default | Running Script](#running-script) to generate the certificates.
+
+#### Different Expiry Date
+
+By default, certificates generated are valid for a period of 3 years. If you need a shorter (or
```

Comment:
Do we do 3 years for CA and client/server certs? Typically, CA certs are longer, like 10 or 20 years. I'm also a fan of 1 year or less for server certs (most of mine are 90 days), but if 3 is what we decided to use, that's fine.

Reply:
I haven't changed this settings from the daedalus installers scripts. It is currently set to `1095` days. I am also in favor of a shorter default expiration period. 1 year seems like a good trade-off. People are still free to change the expiry date as explained if needed.